### PR TITLE
[COST-3788] Sentry: OCPCluster.MultipleObjectsReturned

### PR DESCRIPTION
**Pull request by djnakabaale**
## Jira Ticket

[COST-3788](https://issues.redhat.com/browse/COST-3788)

## Description

This change constraints to the `OCPCluster` table to prevent **_MultipleObjectReturned_** exception

depends on pr https://github.com/project-koku/koku/pull/4367

## Testing

1. Checkout Branch
2. Restart Koku
3. Hit endpoint or launch shell
    1. You should see ...
4. Do more things...

## Notes

...

### CODE REVIEW

File: koku/reporting/provider/ocp/models.py [hunk 1]
```diff
@@ -254,6 +254,7 @@ class Meta:
         """Meta for OCPCluster."""
 
         db_table = "reporting_ocp_clusters"
+        unique_together = ("cluster_id", "cluster_alias", "provider")
```
Comment (myersCody) [inline koku/reporting/provider/ocp/models.py:257]: Do we understand  how multiple rows got added to this?

**Review by myersCody — CHANGES_REQUESTED** [review]
I think we will need to clear our rows where there are already duplicates in the migrations as well. I used the snippet you provided [here](https://github.com/project-koku/koku/pull/4360/files#r1190100695)

to create multiple rows in the db:
```
postgres=# select * from reporting_ocp_clusters;
                 uuid                 |              cluster_id              |    cluster_alias     |             provider_id              
--------------------------------------+--------------------------------------+----------------------+--------------------------------------
 85859ca6-6aa8-4ac0-87fb-85f9a47775a0 | my-ocp-cluster-1                     | Test OCP on AWS      | 8d01f3cc-487c-44f3-a7af-672889585d6d
 bf733439-31bc-422b-9f8f-4995c5d62483 | my-ocp-cluster-2                     | Test OCP on Azure    | 795a12b3-11e1-4732-bc21-3cc857c4b4bf
 8ff86b51-2804-431a-b162-35415b821f3d | test-ocp-gcp-cluster                 | Test OCP on GCP      | a7b1a541-0485-4541-9955-585a20da7870
 c44b0bb6-f090-4018-ac6a-3c0255005ba0 | my-ocp-cluster-3                     | Test OCP on Premises | e5920195-bfc7-40f9-8f3f-2963f54c6a1b
 0d4ca655-e5b0-4ac6-a621-232cd1bacc48 | d6cc251c-8eb6-4a06-954f-c8a3df290169 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
 33a9096e-5a6a-4667-90aa-53b113bb6ceb | d6cc251c-8eb6-4a06-954f-c8a3df290169 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
 39605bb8-6b3f-482f-9a7b-d1e3858108d0 | d6cc251c-8eb6-4a06-954f-c8a3df290169 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
 d3fe1311-1fd5-4001-9962-432b0066d6f7 | 69ea3971-c762-475f-a8cb-498214d873b5 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
 e4fe49ff-114e-41ba-9d76-f35342d233f2 | 69ea3971-c762-475f-a8cb-498214d873b5 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
 a2a84e20-f125-450a-b856-906d0f7bd032 | 69ea3971-c762-475f-a8cb-498214d873b5 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
 58ec81e3-a204-4802-8c67-ec3607dddbad | 3a76f3e4-f9a8-42ea-8bd0-3e122fd2b047 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
 551b9485-d27d-4328-848e-d8963df18a92 | 3a76f3e4-f9a8-42ea-8bd0-3e122fd2b047 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
 31392ffa-52a3-4599-890a-e4c349a397e2 | 3a76f3e4-f9a8-42ea-8bd0-3e122fd2b047 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
```
Then tried to run your migration:
```
  File "/Users/cmyers/.local/share/virtualenvs/koku-WwpPTI0J/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: could not create unique index "reporting_ocp_clusters_cluster_id_cluster_alias_53fc1fd3_uniq"
DETAIL:  Key (cluster_id, cluster_alias, provider_id)=(d6cc251c-8eb6-4a06-954f-c8a3df290169, node_role_test, 165b9403-5545-4546-923d-778ef4e8f822) is duplicated.
```

File: koku/reporting/provider/ocp/models.py (hunk 1 above)
Comment (myersCody) [reply koku/reporting/provider/ocp/models.py:257]: Yeah, I couldn't find anywhere else where we populate this table. Thanks David!

**Comment by djnakabaale** [comment]
> I think we will need to clear our rows where there are already duplicates in the migrations as well. I used the snippet you provided [here](https://github.com/project-koku/koku/pull/4360/files#r1190100695)
> 
> to create multiple rows in the db:
> 
> ```
> postgres=# select * from reporting_ocp_clusters;
>                  uuid                 |              cluster_id              |    cluster_alias     |             provider_id              
> --------------------------------------+--------------------------------------+----------------------+--------------------------------------
>  85859ca6-6aa8-4ac0-87fb-85f9a47775a0 | my-ocp-cluster-1                     | Test OCP on AWS      | 8d01f3cc-487c-44f3-a7af-672889585d6d
>  bf733439-31bc-422b-9f8f-4995c5d62483 | my-ocp-cluster-2                     | Test OCP on Azure    | 795a12b3-11e1-4732-bc21-3cc857c4b4bf
>  8ff86b51-2804-431a-b162-35415b821f3d | test-ocp-gcp-cluster                 | Test OCP on GCP      | a7b1a541-0485-4541-9955-585a20da7870
>  c44b0bb6-f090-4018-ac6a-3c0255005ba0 | my-ocp-cluster-3                     | Test OCP on Premises | e5920195-bfc7-40f9-8f3f-2963f54c6a1b
>  0d4ca655-e5b0-4ac6-a621-232cd1bacc48 | d6cc251c-8eb6-4a06-954f-c8a3df290169 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
>  33a9096e-5a6a-4667-90aa-53b113bb6ceb | d6cc251c-8eb6-4a06-954f-c8a3df290169 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
>  39605bb8-6b3f-482f-9a7b-d1e3858108d0 | d6cc251c-8eb6-4a06-954f-c8a3df290169 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
>  d3fe1311-1fd5-4001-9962-432b0066d6f7 | 69ea3971-c762-475f-a8cb-498214d873b5 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
>  e4fe49ff-114e-41ba-9d76-f35342d233f2 | 69ea3971-c762-475f-a8cb-498214d873b5 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
>  a2a84e20-f125-450a-b856-906d0f7bd032 | 69ea3971-c762-475f-a8cb-498214d873b5 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
>  58ec81e3-a204-4802-8c67-ec3607dddbad | 3a76f3e4-f9a8-42ea-8bd0-3e122fd2b047 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
>  551b9485-d27d-4328-848e-d8963df18a92 | 3a76f3e4-f9a8-42ea-8bd0-3e122fd2b047 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
>  31392ffa-52a3-4599-890a-e4c349a397e2 | 3a76f3e4-f9a8-42ea-8bd0-3e122fd2b047 | node_role_test       | 165b9403-5545-4546-923d-778ef4e8f822
> ```
> 
> Then tried to run your migration:
> 
> ```
>   File "/Users/cmyers/.local/share/virtualenvs/koku-WwpPTI0J/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
>     return self.cursor.execute(sql, params)
> django.db.utils.IntegrityError: could not create unique index "reporting_ocp_clusters_cluster_id_cluster_alias_53fc1fd3_uniq"
> DETAIL:  Key (cluster_id, cluster_alias, provider_id)=(d6cc251c-8eb6-4a06-954f-c8a3df290169, node_role_test, 165b9403-5545-4546-923d-778ef4e8f822) is duplicated.
> ```


@myersCody  Thanks for pointing this out. 

I've split this into two migrations, 
- one to remove duplicates - https://github.com/project-koku/koku/pull/4367
- then apply constraints https://github.com/project-koku/koku/pull/4368

**Pull request closed without merging**